agentstack/main.py

```diff
@@ -199,17 +202,33 @@ def main():
             parser.print_help()
     except Exception as e:
         update_telemetry(telemetry_id, result=1, message=str(e))
-        print(term_color("An error occurred while running your AgentStack command:", "red"))
-        print(e)
-        sys.exit(1)
+        raise e
 
     update_telemetry(telemetry_id, result=0)
 
 
-if __name__ == "__main__":
+def main() -> int:
```

Comment:
`bin/agentstack` calls the app as `sys.exit(main())` so the public `main` function now returns an int for the code and handles all of the logic. 

(`__name__ == '__main__'` doesn't actually get interacted with from the cli)